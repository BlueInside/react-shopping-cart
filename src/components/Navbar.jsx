import PropTypes from 'prop-types';

function Navbar({ links }) {
  return (
    <>
      <nav>
        <ol>
          {links.map((link) => (
            <li key={link.id}>{link.label}</li>
          ))}
        </ol>
      </nav>
    </>
  );
}

Navbar.propTypes = {
  links: PropTypes.array.isRequired,
};

Navbar.defaultProps = {
  links: [],
};
export default Navbar;
