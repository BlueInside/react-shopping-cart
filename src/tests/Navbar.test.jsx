import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  it('renders navbar correctly', () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });

  it('renders links prop (array of links)', () => {
    const testLinks = [
      { id: 'home', label: 'Home', path: '/' },
      { id: 'products', label: 'Products', path: '/products' },
      { id: 'about', label: 'About Us', path: '/about' },
    ];

    render(<Navbar links={testLinks} />);
    const navbar = screen.getByRole('navigation');
    const liElements = screen.getAllByRole('listitem');

    expect(navbar).toBeInTheDocument();
    expect(liElements.length).toBe(testLinks.length);
  });
});
