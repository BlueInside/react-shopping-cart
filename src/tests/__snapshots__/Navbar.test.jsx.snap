// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > renders navbar correctly 1`] = `
<nav
  class="sc-beySPh dJGvpt"
>
  <ol>
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/products"
      >
        Products
      </a>
    </li>
    <li>
      <a
        href="/about"
      >
        Cart
      </a>
    </li>
  </ol>
</nav>
`;
